File: np-lab3/3-1.py
import socket

if __name__ == '__main__':
    fd = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    fd.connect(('inp.zoolab.org', 10301))
    fd.settimeout(0.5)
    
    while True:
        data = fd.recv(1024)
        description = data.decode('utf-8')
        print(description)
        maze_sz_idx = description.find('Note1: Size')
        if (maze_sz_idx == -1):
            print("cannot find index")
            continue
        
        width = int(description[maze_sz_idx+26 : maze_sz_idx+28])
        height = int(description[maze_sz_idx+31 : maze_sz_idx+33])
        
        map = description.find('\n#')
        maze = description[map+1 : map+1+(width+1)*height]        
        maze = maze.split('\n')
        for (w, h) in enumerate(maze):
            for (i, symbol) in enumerate(h):
                if symbol == '*': 
                    start = (w, i)
                elif symbol == 'E':
                    end = (w, i)
        path = ""
        x = start[1] - end[1]
        y = start[0] - end[0]
        if x>0:
            while x>0:
                path += 'A'
                x -= 1
        elif x<0:
            while x<0:
                path += 'D'
                x += 1
        if y>0:
            while y>0:
                path += 'W'
                y -= 1
        elif y<0:
            while y<0:
                path += 'S'
                y += 1
        print("path: ", path)
        path = path + '\n'
        fd.send(path.encode())
        fd.recv(1024)
        flag = fd.recv(1024).decode()[1:]
        print(flag)
        fd.close()
        break  

File: np-lab3/3-1.cpp
#include <netinet/in.h> 
#include <stdio.h> 
#include <stdlib.h> 
#include <string.h> 
#include <cstring> 
#include <sys/socket.h> 
#include <arpa/inet.h>
#include <netdb.h>
#include <unistd.h>
#include <iostream>
using namespace std;

char buff[100000001]; 
char ans[2000];
const char* host = "inp.zoolab.org";
// int maze[11][7];

int main(){
    int socketfd = socket(AF_INET, SOCK_STREAM, 0);
    struct sockaddr_in srv_address, client; 
    if (socketfd == -1) {
        printf("Socket Error\n"); 
        exit(1);
    }
    bzero(&srv_address, sizeof(srv_address));
    srv_address.sin_family = AF_INET;
    srv_address.sin_addr.s_addr = inet_addr(host);
    
    struct hostent *server = gethostbyname(host); 
    if(server == NULL){
        perror("error solving hostname");
        return 1;
    }
    bcopy((char*)server->h_addr, (char*)&srv_address.sin_addr, server->h_length);
    srv_address.sin_port = htons(10301);
    int check_connect = connect(socketfd, (struct sockaddr*)&srv_address, sizeof(srv_address));
    if (check_connect != 0) {
        printf("Connect Error\n"); 
        exit(1); 
    }
    
    while(1){
        bzero(buff, sizeof(buff));
        long long get_ = recv(socketfd, buff, sizeof(buff), 0);
        if(get_ <= 0){
            printf("server closed connect");
            exit(1);
        }
        printf("%s", buff);
        string buffer = buff;      

        int width = 11;
        int length= 7;
        size_t found = buffer.find("\n#");
        int sz = 12*7;
        string maze = buffer.substr(found+1, sz);
        cout << maze
        
       
    }
    
    

     // int maze[maze_len][maze_wid];
        // printf("drawing maze\n");
        // char* wall = strstr(buff, "##");
        // int idx = wall - buff;
        // for(int i=0; i<maze_len; i++){
        //     for(int j=0; j<maze_wid; j++){
        //         maze[i][j] = buff[idx];
        //         idx++;
        //     }
        //     idx++;
        // }
        // for(int i=0; i<maze_len; i++){
        //     for(int j=0; j<maze_wid; j++){
        //         cout << maze[i][j];
        //     }
        // }
        // break;
    

}

